src/struct_mv/struct_matvec.c

```diff
@@ -336,24 +334,18 @@ hypre_StructMatvecCompute( void               *matvec_vdata,
    hypre_IndexRef           start;
 
    HYPRE_Complex            temp;
-   HYPRE_Int                compute_i, i, j, k, si, ssi;
-   HYPRE_Int                csi[UNROLL_MAXDEPTH], vsi[UNROLL_MAXDEPTH];
+   HYPRE_Int                compute_i, i, j, si, se;
+   HYPRE_Int                cnentries, vnentries;
+   HYPRE_Int                centries[nentries], ventries[nentries];
```

Comment:
Should we consider allocating `centries` and `ventries` on the heap instead of using VLAs?

Reply:
The only issue I can see from googling is that VLAs (as written here) are allocated on the stack which is easier to overflow.  Otherwise, VLAs are much faster and produce more readable code, IMO.  Let me do more research on this first before deciding.  Thanks!

Reply:
I will think about this more and consider changing later.  I think it's more important to merge this into recmat right now.  Thanks again!